internal/cli/delete/warehouse.go

```diff
@@ -0,0 +1,60 @@
+package delete
+
+import (
+	goerrors "errors"
+	"fmt"
+
+	"connectrpc.com/connect"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+
+	"github.com/akuity/kargo/internal/cli/client"
+	"github.com/akuity/kargo/internal/cli/option"
+	v1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func newWarehouseCommand(opt *option.Option) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "warehouse [NAME]...",
+		Short: "Delete warehouse by name",
+		Args:  cobra.MinimumNArgs(1),
```

Comment:
In https://github.com/akuity/kargo/pull/1088, I found that the cobra helpers are pretty terse and could use extra help/usage information so someone who ran:
```
kargo delete warehouse
```

would get more helpful information than: `requires at least 1 arg(s), only received 0`

After yours is merged, I'll add another helper to give better user feedback.